svelte-kit should choose new port on EADDRINUSE like sirv

### Describe the problem

I was surprised when I tried to set up the skeleton project and it failed because port 3000 ist already bound.

When I went to check the webserver it was svelte-kit, not sirv, so I have to look up how to change the port.

This shouldn't be that complicated, so I think this should be changed.

```
❯ npm run dev

> ~TODO~@0.0.1 dev /home/coafin/Weidmueller/jws-toolkit
> svelte-kit dev


  SvelteKit v1.0.0-next.137

  local:   http://localhost:3000
  network: not exposed

  Use --host to expose server to other devices on this network



events.js:353
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use 127.0.0.1:3000
    at Server.setupListenHandle [as _listen2] (net.js:1318:16)
    at listenInCluster (net.js:1366:12)
    at GetAddrInfoReqWrap.doListen [as callback] (net.js:1503:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:69:8)
Emitted 'error' event on Server instance at:
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  code: 'EADDRINUSE',
  errno: -98,
  syscall: 'listen',
  address: '127.0.0.1',
  port: 3000
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ~TODO~@0.0.1 dev: `svelte-kit dev`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ~TODO~@0.0.1 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/coafin/.npm/_logs/2021-07-27T06_59_36_177Z-debug.log
```

### Describe the proposed solution

I would like to see the post chosen randomly if the default port is already bound.

### Alternatives considered

Change the port after looking up how :

- run `svelte-kit --help
- fail because PATH
- run `npx svelte-kit --help`
- read
- run `npx svelte-kit dev --help`
- read
- change `svelte-kit dev` to `svelte-kit dev --port <something>` in package.json

### Importance

nice to have

### Additional Information

Low priority and I hope to provide a pull request for a project that I like if I find the time

--- a/.changeset/late-mugs-agree.md
+++ b/.changeset/late-mugs-agree.md
@@ -1,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+Check ports usage in a more cross-platform way in dev server error logging

--- a/packages/kit/src/cli.js
+++ b/packages/kit/src/cli.js
@@ -200,16 +200,21 @@ prog.parse(process.argv, { unknown: (arg) => `Unknown option: ${arg}` });
 
 /** @param {number} port */
 async function check_port(port) {
-	const n = await ports.blame(port);
-
-	if (n) {
+	if (!(await ports.check(port))) {
 		console.log(colors.bold().red(`Port ${port} is occupied`));
 
-		// prettier-ignore
-		console.log(
-			`Terminate process ${colors.bold(n)} or specify a different port with ${colors.bold('--port')}\n`
-		);
-
+		const n = await ports.blame(port); // Only works on systems with `lsof`.
+		if (n) {
+			// prettier-ignore
+			console.log(
+				`Terminate process ${colors.bold(n)} or specify a different port with ${colors.bold('--port')}\n`
+			);
+		} else {
+			// prettier-ignore
+			console.log(
+				`Terminate the process using it or specify a different port with ${colors.bold('--port')}\n`
+			);
+		}
 		process.exit(1);
 	}
 }

--- a/packages/kit/src/cli.js
+++ b/packages/kit/src/cli.js
@@ -200,23 +200,23 @@ prog.parse(process.argv, { unknown: (arg) => `Unknown option: ${arg}` });
 
 /** @param {number} port */
 async function check_port(port) {
-	if (!(await ports.check(port))) {
-		console.log(colors.bold().red(`Port ${port} is occupied`));
-
-		const n = await ports.blame(port); // Only works on systems with `lsof`.
-		if (n) {
-			// prettier-ignore
-			console.log(
-				`Terminate process ${colors.bold(n)} or specify a different port with ${colors.bold('--port')}\n`
-			);
-		} else {
-			// prettier-ignore
-			console.log(
-				`Terminate the process using it or specify a different port with ${colors.bold('--port')}\n`
-			);
-		}
-		process.exit(1);
+	if (await ports.check(port)) {
+		return;
 	}
+	console.log(colors.bold().red(`Port ${port} is occupied`));
+	const n = await ports.blame(port);
+	if (n) {
+		// prettier-ignore
+		console.log(
+			`Terminate process ${colors.bold(n)} or specify a different port with ${colors.bold('--port')}\n`
+		);
+	} else {
+		// prettier-ignore
+		console.log(
+			`Terminate the process occupying the port or specify a different port with ${colors.bold('--port')}\n`
+		);
+	}
+	process.exit(1);
 }
 
 /**

